frontend/public/module/k8s/k8s-models.ts

```diff
@@ -72,3 +80,53 @@ export const modelFor = (ref: K8sResourceKindReference) => {
  * NOTE: This will not work for CRDs defined at runtime, use `connectToModels` instead.
  */
 export const allModels = () => k8sModels;
+
+/**
+ * Use this hook to find the model for resources using only group and plural/kind
+ */
+export const useModelFinder = () => {
+  const models: ImmutableMap<string, K8sKind[]> = useSelector(({ k8s }) =>
+    k8s.getIn(['RESOURCES', 'models']),
+  );
+  const pluralsToModelMap = _.values(models.toJS()).reduce((acc, curr) => {
```

Comment:
You should be able to call `reduce` on the map itself without `_.values`

https://immutable-js.github.io/immutable-js/docs/#/Map/reduce